openassessment/xblock/schema.py

```diff
@@ -76,7 +82,12 @@ def datetime_validator(value):
     Required('feedback_default_text'): utf8_validator,
     Required('submission_start'): Any(datetime_validator, None),
     Required('submission_due'): Any(datetime_validator, None),
-    Required('allow_file_upload'): bool,
+    Required('file_upload_type', default=None): Any(
+        All(utf8_validator, In(VALID_UPLOAD_FILE_TYPE)),
+        None
+    ),
+    'file_type_white_list': utf8_validator,
+    'allow_file_upload': bool,  # Backwards compatibility.
```

Comment:
It would be nice if we keep `allow_file_upload` to it's original location for neat history.